RecoEgamma/EgammaElectronProducers/plugins/alpaka/ElectronNHitSeedAlpakaProducer.cc

```diff
@@ -0,0 +1,227 @@
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaServices/interface/alpaka/AlpakaService.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/global/EDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/SynchronizingEDProducer.h"
+
+// Additional includes for testing / comparing implementations
+#include "PixelMatchingAlgo.h"
+
+using Vector3d = Eigen::Matrix<double, 3, 1>;
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+	class ElectronNHitSeedAlpakaProducer : public global::EDProducer<> {
+		public:
+		ElectronNHitSeedAlpakaProducer(const edm::ParameterSet& pset):
+        	EDProducer(pset),
+			deviceToken_{produces()},
+			size_{pset.getParameter<int32_t>("size")},
+			initialSeedsToken_(consumes(pset.getParameter<edm::InputTag>("initialSeeds"))),
+			beamSpotToken_(consumes(pset.getParameter<edm::InputTag>("beamSpot"))),
+			magFieldToken_(esConsumes()),
+			geomToken(esConsumes())
+			{
+				superClustersTokens_ = consumes(pset.getParameter<edm::InputTag>("superClusters"));
+			}
+
+      	void produce(edm::StreamID sid, device::Event& event, device::EventSetup const& iSetup) const override {
+		//void produce(device::Event& event, device::EventSetup const& iSetup) const override {
+
+			auto vprim_ = event.get(beamSpotToken_).position();
+			GlobalPoint vprim(vprim_.x(), vprim_.y(), vprim_.z());
+			Vector3d vertex{vprim.x(),vprim.y(),vprim.z()};
+
+			// NEW EVENT
+			std::cout<<" -----> NEW EVENT <------- "<<std::endl;
+			// Get MagField ESProduct for comparing & Geom ESProduct
+			// auto const& magField = iSetup.getData(magFieldToken_);
+			// const TrackerGeometry* theG = &iSetup.getData(geomToken);
+			// PropagatorWithMaterial backwardPropagator_ = PropagatorWithMaterial(oppositeToMomentum, 0.000511, &magField);
+
+			std::vector<reco::SuperClusterRef> superClusterRefVec;
+
+			for (auto& superClusRef : event.get(superClustersTokens_))
+				superClusterRefVec.push_back(superClusRef);
```

Comment:
Copy does not seem to be needed
```suggestion
			const std::vector<reco::SuperClusterRef>& superClusterRefVec = event.get(superClustersTokens_);
```